src/cargo/core/compiler/compilation.rs

```diff
@@ -384,6 +375,48 @@ fn fill_rustc_tool_env(mut cmd: ProcessBuilder, unit: &Unit) -> ProcessBuilder {
     cmd
 }
 
+fn get_sysroot_target_libdir(
+    bcx: &BuildContext<'_, '_>,
+) -> CargoResult<HashMap<CompileKind, PathBuf>> {
+    bcx.all_kinds
+        .iter()
+        .map(|&kind| {
+            Ok((
+                kind,
+                bcx.target_data
+                    .get_info(kind)
+                    .ok_or_else(|| {
+                        let target = match kind {
+                            CompileKind::Host => "host".to_owned(),
+                            CompileKind::Target(s) => s.short_name().to_owned(),
+                        };
+
+                        let dependency = bcx
+                            .unit_graph
+                            .iter()
+                            .find_map(|(k, _)| {
+                                if k.kind == kind {
+                                    let summary = k.pkg.manifest().summary();
+
+                                    Some(format!("{} v{}", summary.name(), summary.version()))
+                                } else {
+                                    None
+                                }
+                            })
+                            .unwrap();
+
+                        anyhow::anyhow!(
+                            "could not find specification for target \"{target}\".\n  \
+                            Dependency `{dependency}` requires to build for target \"{target}\"."
```

Comment:
(nit: style)

Oops. That was my oversight. In Cargo by convention we use backtick to quote things. That should be followed here as well I guess.
However, it makes messages inconsistent between rustc and cargo 😆.

```suggestion
                            "could not find specification for target `{target}`.\n  \
                            Dependency `{dependency}` requires to build for target `{target}`."
```